pkg/crc/cluster/cluster.go

```diff
@@ -138,6 +139,41 @@ func StopAndRemovePodsInVM(sshRunner *ssh.SSHRunner) error {
 	return errors.RetryAfter(2, stopAndRemovePods, 2*time.Second)
 }
 
+func AddProxyConfigToCluster(oc oc.OcConfig, proxy *network.ProxyConfig) error {
+	cmdArgs := []string{"patch", "proxy", "cluster", "-p",
+		fmt.Sprintf(`{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s"}}`, proxy.HttpProxy, proxy.HttpsProxy, proxy.NoProxy),
+		"-n", "openshift-config", "--type", "merge"}
+
+	if err := waitForOpenshiftAPIServer(oc); err != nil {
+		return err
+	}
+	if _, stderr, err := oc.RunOcCommand(cmdArgs...); err != nil {
+		return fmt.Errorf("Failed to add proxy details %v: %s", err, stderr)
+	}
+	return nil
+}
+
+// AddProxyToKubeletAndCriO add the systemd drop-in proxy configuration file to the instance,
+// both the services (kubelet and crio) need to be restarted after this change.
+func AddProxyToKubeletAndCriO(sshRunner *ssh.SSHRunner, proxy *network.ProxyConfig) error {
+	proxyTemplate := `[Service]
+Environment=HTTP_PROXY=%s
+Environment=HTTPS_PROXY=%s
+Environment=NO_PROXY=.cluster.local,.svc,10.128.0.0/14,172.30.0.0/16,%s`
+
```

Comment:
Quoting your earlier message on that topic
> @cfergeau there is no doc from openshift side about it, this is taken care by the proxy operator and added automatically but since this status doesn't visible just after making the change in the proxy and need bit time so I used these value as constant because those are the one needed for kubelet and crio both

If these values are added automatically, I'd prefer to let the proxy operator deal with them instead of hardcoding them here, and then need to change them if these values are ever changed to something different on the OpenShift side.

Reply:
>  I'd prefer to let the proxy operator deal with them instead of hardcoding them here, and then need to change them if these values are ever changed to something different on the OpenShift side.

@cfergeau problem is proxy operator is not able to make changes to in the kubelet/crio side, this is job of machine config operator on the node and since for crc this is not possible so we do need to put it here.

Reply:
This kind of context belongs in the commit log, very useful info to have to understand why this was implemented this way.

Reply:
@cfergeau I will add it to commit log and also as part of comment on top of this function.